### ISSUE TITLE
Proposal for a new way of doing things

### ISSUE THREAD

**Issue by chenshuiluke** [PalisadoesFoundation/rfc#3]
# Calico Project Proposal Document

# Current Problems

1. Not that many people are applying.
2. Those that do apply tend to not have the skills required and aren&#39;t necessarily deserving of a 1,500USD stipend.
3. Calico brands itself as a way for students to gain programming experience and to learn from &quot;mentors&quot;, yet we turn away students who really need help in favor of more experienced applicants.
4. Only one student can work on a project at a time.

# Possible Solutions

Not that many people are applying

- Market it more as a summer internship or job
  - Be sure to highlight the financial and time advantages of working on Calico as opposed to a regular summer job:
    - Completely performance based. If you have completed all your tasks, you can spend the remainder of your time doing whatever you want, whether it be vacationing, watching tv, etc. **Due to not having to spend the remainder of your time at work when you have nothing to do, this can lead to an incredibly high &quot;hourly rate&quot;.**
    - The contributions you make to a project can be easily verified and tested by potential employers when put on your resume, due to the projects being open source. This is in contrast to closed source internal software where they have to take your word for it.
    - **Shows you have an interest in practicing your craft and having initiative.**
- Have more interesting and relatable projects. Not that many students are interested in working on a time series database.
  - Students need something they can easily conceptualize and **find use for in their own lives.**
  - This is why JResume had the most applicants last year – almost every young adult has use for a resume generator.
- Possible Project Ideas
  - Create a note-taking app
  - Create a workout app
  - Create a documentation app
  - Make a speed reading app
  - Create a stackable music playlist app
  - Create a daily-&gt;weekly-&gt;monthly review app that uses a wysiwyg editor
  - Make a habit app that allows you to track how much time you spend doing something. For each iteration of a habit, allow the user to make a note on the progress made.
  - Opensource book editor and site
  - Create an accountability app that will make embarrassing posts on social media of you fail to adhere to ur habit schedule
  - Create a dictionary app that uses the Oxford api, the urbandictionary api or the Wikipedia api
  - Create a book reading site that uses both https://github.com/mozilla/pdf.js and https://github.com/futurepress/epub.js
  - Make a HackerNews clone

Those that do apply tend to not have the skills required and aren&#39;t necessarily deserving of a 1,500USD stipend.

Calico brands itself as a way for students to gain programming experience and to learn from &quot;mentors&quot;, yet we turn away students who really need help in favor of more experienced applicants.

Only one student can work on a project at a time.

- Implement a tiered system, where each project can have a mix of beginner, intermediate and advanced applicants.
- Applicants can choose to apply to any of those tiers and will be interviewed and judged based on their chosen tier.
- If an applicant applied to the advanced tier, but is judged to be beginner/intermediate level and there is a space available in the beginner/intermediate tier, then they will be put in that tier and vice versa
- A single project can have a total budget of 1,500 USD to be split amongst the participants of the project
- An advanced participant is entitled to 750 USD (or half the budget)
- An intermediate participant is entitled to 500 USD (or one third of the budget)
- A beginner participant is entitled to 250 USD (or one sixth of the budget)

- There can be a mix of participants of different levels working on a single project, for example:
  - 1 advanced, 1 intermediate, 1 beginner
  - 2 advanced
  - 6 beginners
  - 3 intermediates
  - etc

**Comment by Pr0x1m4** [comment]
Refactoring this issue into smaller ones, all good points

**Comment by Pr0x1m4** [comment]
Interesting points but changing the marketing pitch to a job we run the risk of straying from calico's core/initial purpose of introducing Jamaican students to open source software.

What we could do is highlight the benefits of Calico and increase our prescence online.

**Comment by chenshuiluke** [comment]
well some companies do pay developers to contribute to opensource projects that the company uses, so I don't see it as much of a problem

**Comment by chenshuiluke** [comment]
I like the Calico Academy, but do we have the resources for it? I see no reason why it can't be marketed both as a job and a mentorship experience

**Comment by Pr0x1m4** [comment]
That's the next issue, resources are indeed scarce. The best thing I think we could do are prerecorded videos and challenges (I'd be willing to make some)

The issue I have with marketing it both as a job and work experience is the level of expectation we set. This of course, (I believe) can remedied by tiered applications

**Comment by chenshuiluke** [comment]
I suggest a curated, level-based learning path as opposed to videos, since there are videos and tutorials out there that are probably better than what we can come up with

**Comment by Pr0x1m4** [comment]
Good point, we've had some success with github based challenges, might just expand on what we have

**Comment by palisadoes** [comment]
We need to maintain a long term vision of what we want to achieve. The piecework approach in #4  could deteriorate in paying for contributions in general for Calico and the general GitHub population. It’s risky.

There are a number of issues affecting Calico. Not just marketing.

1. The first is the effectiveness of marketing Calico through the campus student clubs. They tend to be more aware of their abilities and the USD 1,500 isn’t necessarily an incentive as they can earn more on the open market over the Internet doing a wider variety of jobs. As a result, the benefit of experience and cash, isn’t as obvious to them.
2. The second issue is that you can’t plan to get an award to work on Calico. Students need to plan their summer further in advance for various reasons. They arrange trips abroad to work, they get summer jobs and more. By the time Calico is announced this process is well under way, so canceling their plans at the last minute (May) is not a good option. 
3. Awareness seems to be an issue. Not all students are aware, especially outside the clubs and at the post graduate level. 

Possible solutions:

1. We need to find a way to market the program through faculty. This could include having faculty mention the program in their all their under and post grad classes. We could also use Palisadoes to help out in the courses, for example make arrangements through us to have speakers talk to students on different topics for each course. This will help with improving awareness and inspire students to become aware of more background topics, minimizing the need for #5.
2. Consider closing Calico applications in December, announce in January. This will allow students to plan their lives in summer better. It may also make fundraising easier as we’ll have a firm number of awarded students to use for an appeal. The disadvantage is that we could have unfunded students at the beginning of the year, which has  been a cash flow issue in the past.
3. Make Calico a program that runs from January to June. This will provide a better incentive to participate as it could be the equivalent of a part time job. This would also not interfere with the summer plans of students for working in the US to earn money for school. We’ll need to adjust the requirements to work exclusively on Calico over the period. Extending the program over 6 months will help with our fundraising too. It will also give students a longer time to evaluate becoming more familiar with the background topics mentioned in #5. With a longer time-frame we could insist more on students providing better documentation, getting all unit tests done and other sustainability related activities.
4. Rather than having tiers based on ability as in #4 , have two tiers, students and the general public as two separate programs. Calico would continue to be a program for students to work on projects with mentors. The public Calico would have a system of points assigned to each GitHub issue. The first person to have 100 points, say, would get a nominal prize. The benefit would be having code on the Internet visible to employers.
